### PR TITLE
Upgrade react-day-picker

### PR DESCRIPTION
**Pull request by NestorEncinas**
- Rewrite DatePicker & DateRangePicker

JIRA: FET-1061

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8974/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10819/ : FAILURE in 12m 46s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8975/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10821/ : FAILURE in 10m 00s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8976/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10822/ : FAILURE in 10m 13s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8979/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10825/ : FAILURE in 10m 36s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8981/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10829/ : FAILURE in 25m 54s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1739/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8982/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1739/ : UNSTABLE in 39m 13s

**Comment by NestorEncinas** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8986/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10833/ : FAILURE in 10m 25s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1740/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8987/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1740/ : UNSTABLE in 27m 49s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8988/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10834/ : FAILURE in 22m 58s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8990/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10836/ : FAILURE in 20m 17s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8992/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10837/ : SUCCESS in 20m 06s

**Comment by NestorEncinas** [comment]
extended check cypress

**Comment by yenkins** [comment]
# ✨ Cypress test results ✨

## 😢 Some tests are failing. Consider fixing them.

|Status|Name|Tests|Failures|Time|
|:--:|:--|:--:|:--:|--:|
|✅|**pivotTableRemoveAllFromMutipleAggregation.spec.ts**|1|-|11.8450s|
|✅|**attributeFilterButton.spec.ts**|9|-|37.4860s|
|✅|**dashboard.spec.ts**|9|-|56.4740s|
|✅|**pivotTableSizingReset.spec.ts**|15|-|46.4160s|
|✅|**pivotTableAggregationsMenuAndAddTotal.spec.ts**|5|-|26.8390s|
|✅|**headlineChartDrilling.spec.ts**|2|-|8.5220s|
|✅|**attributeFilterConfig.spec.ts**|3|-|33.3950s|
|✅|**parentChildFiltersWithPlaceholders.spec.ts**|3|-|24.4910s|
|❌|**[🎥 dateFilter.spec.ts](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/261/artifact/libs/sdk-ui-tests-e2e/cypress/videos/)**|9|<span style="color: red">3</span>|17.0740s|
|✅|**attributeFilterButtonWithSelection.spec.ts**|2|-|13.2790s|
|✅|**dashboardFiltering.spec.ts**|2|-|27.9980s|
|✅|**attributeFilterButtonParentChild.spec.ts**|3|-|19.8470s|
|✅|**barChartDrilling.spec.ts**|4|-|11.2980s|
|✅|**pivotTableRemoveOneAggregation.spec.ts**|1|-|12.3120s|
|✅|**pivotTableSizing.spec.ts**|1|-|10.3280s|
|✅|**pivotTableRemoveOneFromMutipleAggregation.spec.ts**|1|-|12.6730s|
|✅|**drilling.spec.ts**|2|-|21.6240s|

You can find full report [here](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/261/).

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8993/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/261/ : UNSTABLE in 17m 25s

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8995/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10840/ : SUCCESS in 20m 06s

**Comment by NestorEncinas** [comment]
extended test

**Comment by NestorEncinas** [comment]
extended check cypress

**Comment by yenkins** [comment]
# ✨ Cypress test results ✨

## 😢 Some tests are failing. Consider fixing them.

|Status|Name|Tests|Failures|Time|
|:--:|:--|:--:|:--:|--:|
|✅|**headlineChartDrilling.spec.ts**|2|-|8.6150s|
|❌|**[🎥 dateFilter.spec.ts](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/262/artifact/libs/sdk-ui-tests-e2e/cypress/videos/)**|9|<span style="color: red">1</span>|19.8060s|
|✅|**pivotTableRemoveOneAggregation.spec.ts**|1|-|13.7480s|
|✅|**dashboardFiltering.spec.ts**|2|-|24.6940s|
|✅|**attributeFilterButton.spec.ts**|9|-|34.7390s|
|✅|**attributeFilterButtonWithSelection.spec.ts**|2|-|11.0740s|
|✅|**attributeFilterConfig.spec.ts**|3|-|32.3320s|
|✅|**pivotTableAggregationsMenuAndAddTotal.spec.ts**|5|-|26.3370s|
|✅|**parentChildFiltersWithPlaceholders.spec.ts**|3|-|23.6200s|
|✅|**attributeFilterButtonParentChild.spec.ts**|3|-|19.3140s|
|✅|**dashboard.spec.ts**|9|-|53.6750s|
|✅|**pivotTableRemoveAllFromMutipleAggregation.spec.ts**|1|-|11.8950s|
|✅|**barChartDrilling.spec.ts**|4|-|11.8300s|
|✅|**drilling.spec.ts**|2|-|20.5710s|
|✅|**pivotTableRemoveOneFromMutipleAggregation.spec.ts**|1|-|12.9300s|
|✅|**pivotTableSizingReset.spec.ts**|15|-|44.6360s|
|✅|**pivotTableSizing.spec.ts**|1|-|9.8650s|

You can find full report [here](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/262/).

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8998/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/262/ : UNSTABLE in 14m 57s

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8997/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1742/ : FAILURE in 27m 39s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8996/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10842/ : SUCCESS in 29m 54s

File: libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateRangePicker.tsx [hunk 1]
```diff
@@ -42,66 +107,90 @@ class DateRangePickerComponent extends React.Component<IDateRangePickerProps & W
...
         return (
             <>
                 {isTimeEnabled ? (
-                    <div className="gd-date-range-picker datetime s-date-range-picker">
-                        <label>{intl.formatMessage({ id: "filters.from" })}</label>
-                        {FromField}
-                        <label>{intl.formatMessage({ id: "filters.to" })}</label>
-                        {ToField}
-                    </div>
+                    <>
```
Comment (pbenes) [inline libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateRangePicker.tsx:175]: It seems to me that we would not need `<>` for this particular case. We would perhaps also do not need them in the second part of the condition, if 
```
{this.state.isOpen && DatePicker}
```
would be placed directly below {ToField} like in this part. 

Just curious, is there some reason for having it differently in each part of the condition?

File: libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateTimePicker.tsx [hunk 2]
```diff
@@ -1,51 +1,87 @@
...
+        //
+        // it's also necessary to parse only when the input string fully matches with the desired format
+        // to make sure that the picker input is not overwritten in the middle of writing.
+        // e.g, let's consider a case where dateFormat is "dd/MM/yyyy" and the DayPickerInput has already been filled with a valid string "13/09/2020",
+        // then an user wants to change only the month "13/09/2020" -> "13/11/2020" by removing "09" and typing "11".
+        // in such case the parsing should wait until the user completes typing "11" (otherwise if parsing is done right after the first "1" is typed,
+        // the cursor automatically moves to the end of the string in the middle of writing, causing a bad experience for the user).
+        if (
+            isValid(parsedDate) &&
+            parsedDate.getFullYear() >= 1000 &&
```
Comment (pbenes) [inline libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateTimePicker.tsx:34]: just curious, why 1000?

File: libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateTimePicker.tsx [hunk 3]
```diff
@@ -92,23 +136,26 @@ const DateTimePickerComponent = React.forwardRef<DayPickerInput, DateTimePickerC
                         value={getPlatformStringFromDate(value)}
                     />
                 ) : (
-                    <DateRangePickerInputField
+                    <div
                         className={cx(
-                            `s-date-range-picker-date`,
```
Comment (pbenes) [inline libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateTimePicker.tsx:97]: I'm surprised I don't see this used in any e2e test, so :+1: for removing.

File: libs/sdk-ui-filters/src/DateFilter/tests/utils.ts [hunk 4]
```diff
@@ -5,6 +5,9 @@ import { CommonWrapper, ShallowWrapper, ReactWrapper } from "enzyme";
 export const childGetter = (selector: any) => (component: ShallowWrapper | ReactWrapper) =>
     component.find(selector);
 
+export const nodeChildGetter = (selector: any) => (component: ShallowWrapper | ReactWrapper) =>
```
Comment (pbenes) [inline libs/sdk-ui-filters/src/DateFilter/tests/utils.ts:8]: what about `firstChildGetter` ?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8999/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10843/ : SUCCESS in 21m 01s

**Comment by yenkins** [comment]
# ✨ Cypress test results ✨

## 😢 Some tests are failing. Consider fixing them.

|Status|Name|Tests|Failures|Time|
|:--:|:--|:--:|:--:|--:|
|✅|**barChartDrilling.spec.ts**|4|-|12.9860s|
|✅|**pivotTableRemoveOneFromMutipleAggregation.spec.ts**|1|-|14.3610s|
|❌|**[🎥 dateFilter.spec.ts](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/263/artifact/libs/sdk-ui-tests-e2e/cypress/videos/)**|9|<span style="color: red">1</span>|21.4290s|
|✅|**attributeFilterButtonWithSelection.spec.ts**|2|-|12.5780s|
|✅|**dashboard.spec.ts**|9|-|57.0550s|
|✅|**pivotTableAggregationsMenuAndAddTotal.spec.ts**|5|-|27.0310s|
|✅|**headlineChartDrilling.spec.ts**|2|-|8.7030s|
|✅|**pivotTableSizing.spec.ts**|1|-|10.7050s|
|✅|**pivotTableSizingReset.spec.ts**|15|-|45.7010s|
|✅|**attributeFilterButtonParentChild.spec.ts**|3|-|19.6410s|
|✅|**dashboardFiltering.spec.ts**|2|-|26.8590s|
|✅|**attributeFilterButton.spec.ts**|9|-|36.9120s|
|✅|**parentChildFiltersWithPlaceholders.spec.ts**|3|-|24.9330s|
|✅|**drilling.spec.ts**|2|-|23.2290s|
|✅|**pivotTableRemoveOneAggregation.spec.ts**|1|-|13.3130s|
|✅|**attributeFilterConfig.spec.ts**|3|-|33.4590s|
|✅|**pivotTableRemoveAllFromMutipleAggregation.spec.ts**|1|-|12.9520s|

You can find full report [here](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/263/).

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9006/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/263/ : UNSTABLE in 17m 09s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9005/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10849/ : SUCCESS in 22m 44s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9012/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10855/ : SUCCESS in 21m 19s

File: libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateRangePicker.tsx (hunk 1 above)
Comment (NestorEncinas) [reply libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateRangePicker.tsx]: For time enabled, need to see which RangePicker is selected, so is attached to the correct input.

File: libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateTimePicker.tsx (hunk 2 above)
Comment (NestorEncinas) [reply libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateTimePicker.tsx:34]: Not quite sure, was already there. Guess is some other condition to be sure that the input has correct value before starting with the calculations.

Comment (pbenes) [reply libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateTimePicker.tsx:34]: :+1:

**Comment by NestorEncinas** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9016/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10858/ : SUCCESS in 19m 22s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1746/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9017/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1746/ : UNSTABLE in 26m 58s

**Comment by NestorEncinas** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9024/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10862/ : SUCCESS in 26m 27s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1749/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9025/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1749/ : SUCCESS in 37m 24s

**Review by pbenes — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2796

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2575/ : SUCCESS in 5m 41s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2874/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1705/ : SUCCESS in 20m 24s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10877/ : SUCCESS in 21m 26s

**Pull request merged**